### ISSUE TITLE
Add a ranking option for the projects

### ISSUE THREAD

**Issue by SamyMe** [fosjump/fosjump#4]
Thanks to the **/r/opensource** subreddit community, the number of project listed by [FOS Jump](https://fosjump.github.io) already exceeded a dozen.
It would be great, especially if the number keeps increasing at this rate, to add a ranking option. I am not a web-dev (I do mostly AI actuall) so I might need some help.

For each project there could be a little + button, arrow up like in reddit or stackoverflow (the arrow down is not very important though).
![Something Like This](https://i.imgur.com/825uwd0.png)

To run the project:
- Fork it
- install the requirements `pip install -r requirements.txt`
- Modify what you want to then run `python generate.py`
- If all is right, submit your PR

We need to:
- [x] Propose some ideas
- [ ] Chose the simplest 
- [ ] Implement it

**Comment by Endi1** [comment]
This is a static site, right? If you want to keep it that way (which I think you do) this will have to be implemented on the frontend. Problem is, problem is, we need to store the data (number of votes for example) somewhere and this cannot be done on the frontend.

**Comment by SamyMe** [comment]
Yes it's static but I don't necessarily want to keep it that way (for the moment, it does the job). 

I don't have JS experience, but I was wondering if there was a way for modifying a file in the repo? (but now that I'm saying it, seems a little undo-able).

**Comment by Endi1** [comment]
I don't believe it's doable on the frontend.

**Comment by SamyMe** [comment]
What about putting that file in a different server (which I have)? Do you think there could be a way to do so?

**Comment by Endi1** [comment]
If you have access to a server you can run a database there and make AJAX calls from the static website (through Angular or React or even just jQuery) to an API connected to that database. This would be ideal.

If you can't run a db there, you will still need to have an API written for the backend (PHP, Python, Node.js or whatever), there needs to be a script running in the backend to be able to modify the file, JavaScript running in the browser can't do that because it's not secure.

**Comment by SamyMe** [comment]
That sounds great ! I think I will investigate that. If you have time to propose some code please feel free to do so.

**Comment by Endi1** [comment]
Sure thing! One more thing that I just thought of. For the database you can use [sqlite](https://www.sqlite.org/). It's pretty lightweight and it can be stored as a single file.

**Comment by simmac** [comment]
What you have to think about is vote validation. Do you care if users could vote on a single project over and over again?
If yes, how do you want to protect the voting system against abuse? Cookies? This is easily circumventable.
IP-Adresses? This is a privacy issue.
Accounts? Sounds like overkill.

**Comment by SamyMe** [comment]
Accounts are too much for the moment indeed. And IPs would've been the easiest choice, but yeah, I don't know about the privacy issue ... I need to think about that.
